### PR TITLE
[chore] tweak init command

### PR DESCRIPTION
**Pull request by fabio-ivona**
This PR will improve `vendor/bin/pest --init` command: 

**for Laravel projects**

it will require `pestphp/pest-plugin-laravel` package and run `artisan pest:install`

**for plain PHP projects**

it will improve stubs: 

image

### CODE REVIEW

File: src/Plugins/Init.php [hunk 1]
```diff
@@ -98,4 +106,68 @@ private function init(): void
...
     }
+
+    private function initLaravel(): int
+    {
+        View::render('components.badge', [
+            'type' => 'INFO',
+            'content' => 'Laravel installation detected, pest-plugin-laravel will be installed.',
+        ]);
+
+        exec('composer require pestphp/pest-plugin-laravel 2.x-dev', result_code: $result);
```
Comment (nunomaduro) [inline src/Plugins/Init.php]: Can you check the code we are using on Jetstream / Breeze to run these sort of commands?

File: src/Plugins/Init.php [hunk 2]
```diff
@@ -98,4 +106,68 @@ private function init(): void
...
+            'left' => 'Pest',
+            'right' => 'Installed in Laravel',
+        ]);
+
+        View::render('components.new-line');
+
+        return 0;
+    }
+
+    private function isLaravelInstalled(): bool
```
Comment (nunomaduro) [inline src/Plugins/Init.php:143]: Add a description explaining what's is this doing.

File: src/Plugins/Init.php [hunk 3]
```diff
@@ -98,4 +106,68 @@ private function init(): void
 
         exit(0);
     }
+
+    private function initLaravel(): int
+    {
+        View::render('components.badge', [
```
Comment (nunomaduro) [inline src/Plugins/Init.php]: Remove this.

File: src/Plugins/Init.php [hunk 4]
```diff
@@ -98,4 +106,68 @@ private function init(): void
...
+            'type' => 'INFO',
+            'content' => 'Laravel installation detected, pest-plugin-laravel will be installed.',
+        ]);
+
+        exec('composer require pestphp/pest-plugin-laravel 2.x-dev', result_code: $result);
+
+        /** @var int $result */
+        if ($result > 0) {
```
Comment (nunomaduro) [inline src/Plugins/Init.php:123]: If the installation of `pestphp/pest-plugin-laravel` fails, you should display the error output, using symfony process console error out and exit.

File: src/Plugins/Init.php [hunk 5]
```diff
@@ -98,4 +106,68 @@ private function init(): void
...
+            View::render('components.badge', [
```
Comment (nunomaduro) [inline src/Plugins/Init.php]: Remove this.

File: src/Plugins/Init.php [hunk 6]
```diff
@@ -98,4 +106,68 @@ private function init(): void
...
+                'type' => 'ERROR',
+                'content' => 'Something went wrong while installing pest-plugin-laravel package. Please refer the above output for more info.',
+            ]);
+
+            return $result;
+        }
+
+        View::render('components.badge', [
```
Comment (nunomaduro) [inline src/Plugins/Init.php]: Remove this.

File: src/Plugins/Init.php [hunk 7]
```diff
@@ -98,4 +106,68 @@ private function init(): void
...
+            'type' => 'INFO',
+            'content' => 'Running artisan command to install Pest.',
+        ]);
+
+        $app = require $this->testSuite->rootPath.'/bootstrap/app.php';
```
Comment (nunomaduro) [inline src/Plugins/Init.php]: Remove this.

File: src/Plugins/Init.php [hunk 8]
```diff
@@ -98,4 +106,68 @@ private function init(): void
...
+        /** @phpstan-ignore-next-line */
+        $app->make(Kernel::class)->bootstrap();
```
Comment (nunomaduro) [inline src/Plugins/Init.php]: Remove this.

File: src/Plugins/Init.php [hunk 9]
```diff
@@ -98,4 +106,68 @@ private function init(): void
...
+
+        /** @phpstan-ignore-next-line */
+        $result = Process::run('php artisan pest:install --no-interaction');
```
Comment (nunomaduro) [inline src/Plugins/Init.php]: Use Symfony's process class, so we don't have to boot Laravel.

File: src/Plugins/Init.php [hunk 10]
```diff
@@ -98,4 +106,68 @@ private function init(): void
...
+
+        if ($result->failed()) {
```
Comment (nunomaduro) [inline src/Plugins/Init.php]: No need, just display the regular output of the command.

File: src/Plugins/Init.php [hunk 11]
```diff
@@ -98,4 +106,68 @@ private function init(): void
...
+            $this->output->writeln($result->errorOutput());
+
+            View::render('components.badge', [
```
Comment (nunomaduro) [inline src/Plugins/Init.php]: Remove this.

File: src/Plugins/Init.php [hunk 12]
```diff
@@ -98,4 +106,68 @@ private function init(): void
...
+                'type' => 'ERROR',
+                'content' => 'Something went wrong while installing Pest in laravel. Please refer the above output for more info.',
+            ]);
+
+            return $result->exitCode();
+        }
+
+        $this->output->writeln($result->output());
+
+        View::render('components.two-column-detail', [
```
Comment (nunomaduro) [inline src/Plugins/Init.php]: Remove this.

File: src/Plugins/Init.php [hunk 13]
```diff
@@ -98,4 +106,68 @@ private function init(): void
...
+            'left' => 'pest-plugin-laravel',
+            'right' => 'Installed',
+        ]);
+
+        View::render('components.two-column-detail', [
```
Comment (nunomaduro) [inline src/Plugins/Init.php]: Remove this.

**Review by nunomaduro — CHANGES_REQUESTED** [review]

File: src/Plugins/Init.php (hunk 5 above)
Comment (fabio-ivona) [reply src/Plugins/Init.php]: the updated code already prints error output, so no need to do here too

File: src/Plugins/Init.php (hunk 4 above)
Comment (fabio-ivona) [reply src/Plugins/Init.php:123]: the updated code already prints error output, so no need to do here too

**Pull request merged**